app/uk/gov/hmrc/valuationofficeagencycontactfrontend/controllers/TellUsMoreController.scala

```diff
@@ -0,0 +1,100 @@
+import uk.gov.hmrc.valuationofficeagencycontactfrontend.{FrontendAppConfig, Navigator}
+import uk.gov.hmrc.valuationofficeagencycontactfrontend.forms.TellUsMoreForm
+import uk.gov.hmrc.valuationofficeagencycontactfrontend.identifiers.TellUsMoreId
+import uk.gov.hmrc.valuationofficeagencycontactfrontend.models.{Mode, TellUsMore}
+import uk.gov.hmrc.valuationofficeagencycontactfrontend.utils.UserAnswers
+import uk.gov.hmrc.valuationofficeagencycontactfrontend.views.html.tellUsMore
+
+import scala.concurrent.Future
+
+class TellUsMoreController @Inject()(appConfig: FrontendAppConfig,
+                                                  override val messagesApi: MessagesApi,
+                                                  dataCacheConnector: DataCacheConnector,
+                                                  navigator: Navigator,
+                                                  getData: DataRetrievalAction,
+                                                  requireData: DataRequiredAction) extends FrontendController with I18nSupport {
+
+  def councilTaxKey(answers: UserAnswers): Option[String] = {
+
+    answers.councilTaxSubcategory match{
+      case Some("council_tax_home_business") => Some("councilTaxSubcategory.council_tax_home_business")
+      case Some("council_tax_change") => Some("councilTaxSubcategory.council_tax_change")
+      case Some("council_tax_assess") => Some("councilTaxSubcategory.council_tax_assess")
+      case Some("council_tax_other") => Some("councilTaxSubcategory.council_tax_other")
+      case _ => None
+    }
+  }
+
+  def businessRatesKey(answers: UserAnswers): Option[String] = {
+
+    answers.businessRatesSubcategory match{
+      case Some("business_rates_rateable_value") => Some("businessRatesSubcategory.business_rates_rateable_value")
+      case Some("business_rates_moved_property") => Some("businessRatesSubcategory.business_rates_moved_property")
+      case Some("business_rates_other") => Some("businessRatesSubcategory.business_rates_other")
+      case _ => None
+    }
+  }
+
+  def enquiryKey(answers: UserAnswers): Option[String] = {
+
+    answers.enquiryCategory match{
+      case Some("council_tax") => councilTaxKey(answers)
+      case Some("business_rates") => businessRatesKey(answers)
+      case _ => None
+    }
+  }
+
+  def onPageLoad(mode: Mode) = (getData andThen requireData) {
+    implicit request =>
+      val preparedForm = request.userAnswers.tellUsMore match {
+        case None => TellUsMoreForm()
+        case Some(value) => TellUsMoreForm().fill(value)
+      }
+
+      enquiryKey(request.userAnswers) match{
+        case Some(key) =>  Ok(tellUsMore(appConfig, preparedForm, mode, key))
+        case None => {
+          Logger.warn("Navigation for Tell us more page reached without selection of enquiry by controller")
+          throw new RuntimeException("Navigation for Tell us more page reached without selection of enquiry by controller")
```

Comment:
Could we include more information in the logged comment - e.g. was it the business branch that failed or the council tax?